src/main/java/views/DashboardForm.java

```diff
@@ -0,0 +1,253 @@
+        allSnippetsButton.addActionListener(e -> handleAllSnippetsButton());
+
+        addSnippetButton.setIcon(getScaledImageIcons(new ImageIcon("src/main/assets/add_snippet_icon.png"),120, 120));
+        setupDashboadButton(addSnippetButton);
+        addSnippetButton.addActionListener(e -> handleAllSnippetsButton());
+
+        settingsButton.setIcon(getScaledImageIcons(new ImageIcon("src/main/assets/settings_icon.png"),120, 120));
+        setupDashboadButton(settingsButton);
+        settingsButton.addActionListener(e -> handleSettingsButtonClicked());
+
+        for(Snippet snippet : Globals.snippetHelper.getSnippetsOrderByDateDescending()){
+            recentlyAddedListModel.addElement(snippet);
+        }
+
+        //Setting list view and selection mode
+        recentlyAddedListView.setCellRenderer(new RecentlyAddedCellRenderer());
+        recentlyAddedListView.setModel(recentlyAddedListModel);
+        recentlyAddedListView.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+
+        recentlyAddedListView.addMouseListener(new MouseAdapter() {
+            public void mouseClicked(MouseEvent evt) {
+                JList list = (JList)evt.getSource();
+                if (evt.getClickCount() == 2) {
+
+                    int index = list.locationToIndex(evt.getPoint());
+
+                    JPanel panel = new JPanel();
+                    JDialog optionsDialog = new JDialog((JFrame)null, "Select an option");
+
+                    panel.setBorder(BorderFactory.createEmptyBorder(30,30,30,30));
+                    panel.setLayout(new GridLayout(1,3, 20, 20));
+                    JButton editButton = new JButton("Edit");
+                    JButton copyButton = new JButton("Copy");
+                    JButton deleteButton = new JButton("Delete");
+
+                    editButton.setIcon(getScaledImageIcons(new ImageIcon("src/main/assets/edit_icon.png"),50, 50));
+                    copyButton.setIcon(getScaledImageIcons(new ImageIcon("src/main/assets/copy_icon.png"),50, 50));
+                    deleteButton.setIcon(getScaledImageIcons(new ImageIcon("src/main/assets/delete_icon.png"),50, 50));
+
+                    setupOptionsButton(editButton);
+                    setupOptionsButton(copyButton);
+                    setupOptionsButton(deleteButton);
+
+                    //Add eventListeners
+                    deleteButton.addActionListener(e -> handleDeleteButtonPressed(optionsDialog, index));
+                    copyButton.addActionListener(e -> handleCopyButtonPressed(optionsDialog, index));
+
+                    panel.add(editButton);
+                    panel.add(copyButton);
+                    panel.add(deleteButton);
+
+
+                    optionsDialog.getContentPane().add(panel);
+                    setOptionsDialogSettings(optionsDialog);
+
+
+
+                    optionsDialog.setVisible(true);
+
+                }
+            }
+        });
+    }
```

Comment:
It is, indeed, very long. I would try to split it up, but i understand how GUI programming leads to long methods like these.